zapcore/entry.go

```diff
@@ -147,6 +147,7 @@ type Entry struct {
 	LoggerName string
 	Message    string
 	Caller     EntryCaller
+	Function   string
```

Comment:
I had the same thought while I was implementing this. I think I was confused in the past and so I ended up putting `Function` as a log entry instead of being part of `EntryCaller`. 

Now I think again, it is reasonable to put `Function` as part of `EntryCaller`.